lib/relay.js

```diff
@@ -197,15 +251,31 @@ function responseHandler(filterRules, config) {
 
       logger.debug(logContext, 'sending websocket request over HTTP connection');
 
-      request({
+      var req = {
         url: result.url,
         headers: headers,
         method,
         body,
         agentOptions: {
           ca: config.caCert, // Optional CA cert
         },
-      }, (error, response, body) => {
+      };
+
+      // check if this is a streaming request for binary data
+      if (streamingID) {
+        logger.debug(logContext, 'serving binary stream request');
+
+        req.encoding = null; // no encoding for binary data
+
+        request(req).on('data', chunk => {
```

Comment:
As mentioned above - we pipe all chunks of the response directly to the broker server which then forwards everything directly to the consuming service.